### ISSUE TITLE
A `SQLITE_CONSTRAINT: FOREIGN KEY constraint failed` error shows up on startup

### ISSUE THREAD

**Issue by jamietanna** [mend/renovate-ce-ee#325]
Is this error expected when starting up Renovate CE fresh?

```
ERROR: Error adding repository                                                                                                                                                                             
       "driver": "sqlite",                                                                                                                                                                                 
       "err": {                                                                                                                                                                                            
         "errno": 19,                                                                                                                                                                                      
         "code": "SQLITE_CONSTRAINT",                                                                                                                                                                      
         "message": "SQLITE_CONSTRAINT: FOREIGN KEY constraint failed",                                                                                                                                    
         "stack": "Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed"                                                                                                                                
       },                                                                                                                                                                                                  
       "installation": "tanna.dev/jvt.me-examples/go-mod-fork-gotcha"                                                                                                                                      
ERROR: Error adding repository                                                                                                                                                                             
       "driver": "sqlite",                                                                                                                                                                                 
       "err": {                                                                                                                                                                                            
         "errno": 19,                                                                                                                                                                                      
         "code": "SQLITE_CONSTRAINT",                                                                                                                                                                      
         "message": "SQLITE_CONSTRAINT: FOREIGN KEY constraint failed",                                                                                                                                    
         "stack": "Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed"                                                                                                                                
       },                                                                                                                                                                                                  
       "installation": "tanna.dev/jvt.me-examples/buildkite-dynamic-example"                                                                                                                               
 INFO: Starting scheduler                                                                                                                                                                                  
       "scheduler": "app sync scheduler",                                                                                                                                                                  
       "cronTime": "0 0,4,8,12,16,20 * * *",                                                                                                                                                               
       "localTime": "Thu Sep 28 2023 18:44:17 GMT+0000 (Coordinated Universal Time)"                                                                                                                       
 INFO: Scheduler started                                                                                                                                                                                   
       "scheduler": "app sync scheduler"                            
```

It's unclear as to which of the foreign keys may be at fault, based on schema:

```sql
CREATE TABLE repo
(
    uuid      TEXT PRIMARY KEY, -- uuid(full_name, NAMESPACE)
    org_id    TEXT NOT NULL,
    full_name TEXT NOT NULL,
    FOREIGN KEY (org_id) REFERENCES org (uuid) ON DELETE CASCADE ON UPDATE CASCADE
);
CREATE TABLE job_queue
(
    id         integer PRIMARY KEY AUTOINCREMENT,
    repo_id    TEXT      NOT NULL UNIQUE,
    reason     TEXT      NOT NULL,
    priority   INTEGER   NOT NULL,
    attempts   INTEGER   NOT NULL DEFAULT 0,
    added_at   TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
    started_at TIMESTAMP,
    last_alive TIMESTAMP,
    FOREIGN KEY (repo_id) REFERENCES repo (uuid) ON DELETE CASCADE ON UPDATE CASCADE
);
```

This is being run on a GitLab installation with [`Dockerfile` defined in this MR](https://gitlab.com/tanna.dev/renovate-runner/-/merge_requests/1)

**Comment by nabeelsaabna** [comment]
Is it possible to get the logs from the start until the error ?

**Comment by jamietanna** [comment]
Sure, this is what I've got:

```
 INFO: Validating terms of service agreement           
       "input": "y"                                    
 INFO: Validating Mend license key                                                                                                                                
 INFO: This is a CORPORATE Mend Renovate license, registered to: Www.jvt.me. It was registered by Jamie Tanna (<redacted>)
 INFO: Using GitLab endpoint                                                                                                                                      
       "endpoint": "https://gitlab.com/api/v4/"        
 INFO: Using GitLab endpoint: https://gitlab.com/api/v4/
 INFO: Authenticated as GitLab user                    
       "username": "jamietanna_bot"                    
ERROR: Error adding repository                         
       "driver": "sqlite",                                                                                                                                        
       "err": {                                        
         "errno": 19,                                  
         "code": "SQLITE_CONSTRAINT",                  
         "message": "SQLITE_CONSTRAINT: FOREIGN KEY constraint failed",   
         "stack": "Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed"
       },                                              
       "installation": "tanna.dev/jvt.me-examples/go-mod-fork-gotcha"
ERROR: Error adding repository                         
       "driver": "sqlite",                                                                                                                                        
       "err": {                                        
         "errno": 19,                                  
         "code": "SQLITE_CONSTRAINT",                                                                                                                             
         "message": "SQLITE_CONSTRAINT: FOREIGN KEY constraint failed",
         "stack": "Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed"
       },                                              
       "installation": "tanna.dev/jvt.me-examples/buildkite-dynamic-example"
 INFO: Starting scheduler                                                                                                                                         
       "scheduler": "app sync scheduler",              
       "cronTime": "0 0,4,8,12,16,20 * * *",           
       "localTime": "Thu Sep 28 2023 19:30:38 GMT+0000 (Coordinated Universal Time)"
 INFO: Scheduler started                                                                                                                                          
       "scheduler": "app sync scheduler"               
 INFO: Starting scheduler                              
       "scheduler": "job scheduler",                   
       "cronTime": "0 * * * *",                                                                                                                                   
       "localTime": "Thu Sep 28 2023 19:30:38 GMT+0000 (Coordinated Universal Time)"
```

I've tried upping to `LOG_LEVEL=debug` but that doesn't show anything else at this point

**Comment by nabeelsaabna** [comment]
`LOG_LEVEL=debug` and `LOG_FORMAT=json` should work fine 🤔

**Comment by jamietanna** [comment]
I'm wondering if it's because it's in the Community Edition code, not pure Renovate, so slightly different loggers and setup?

I'm not sure if I'm legally allowed to deobfuscate the code to try and debug any more than that.

**Comment by nabeelsaabna** [comment]
well. it's actually the exact same logger as Renovate OSS

**Comment by jamietanna** [comment]
Hmm. Are there any other configurations or environment variables we can use to get more info out of SQLite or Renovate CE?

**Comment by PhilipAbed** [comment]
@jamietanna 
try to rename so you dont have double slash in the name
tanna.dev/jvt.me-examples/go-mod-fork-gotcha

i think we have a problem when trying to split the org from the repo using `{{org/repo}}.split('/')[0]`

**Comment by jamietanna** [comment]
Oh, if so that may a bug on the Renovate CE side then, http://gitlab.com/tanna.dev/jvt.me-examples/go-mod-fork-gotcha is a valid GitLab project URL, as GitLab allows nesting of groups (organisations)

**Comment by PhilipAbed** [comment]
after investigating, i think that is the issue, but i wanted you to confirm, are these the only two repositories that have a problem?
if yes, check if other repos have those slashes, to confirm my suspicion, 
im opening an internal task to resolve this asap on renovate-CE

**Comment by jamietanna** [comment]
Yep, looks like it is that :raised_hands:

**Comment by PhilipAbed** [comment]
i'm in gitlab server, and it doesn't allow me to name any repository like that. gives me errors such as 

renaming repo:
![image](https://github.com/mend/renovate-ce-ee/assets/42116482/dd91d068-e94d-4e35-aa89-0b905b2641ca)

renaming user:
![image](https://github.com/mend/renovate-ce-ee/assets/42116482/25616a80-26e1-4102-acfd-6bf2ba786222)

seems its only a gitlab.com behaviour

**Comment by jamietanna** [comment]
I don't think a rename would do it - if you create a new Group (or use an existing one), and then under it create a new Group, then a repo, that should give you the same behaviour

**Comment by PhilipAbed** [comment]
opened a task for supporting subgroups. thanks for reporting!

**Comment by Reamer** [comment]
We also use many subgroups in GitLab. @jamietanna Thank you for your detailed bug report. I hope for a quick solution to update to 6.x.

**Comment by NINNiT** [comment]
We just tried to migrate from OSS to CE today. Looking forward to a usable 6.x, as we also use a good amount of subgroups.

**Comment by nabeelsaabna** [comment]
fixed in 6.2.0
